e2e/testcases/sync_ordering_test.go

```diff
@@ -662,6 +666,114 @@ func TestDependencyWithReconciliation(t *testing.T) {
+		ntopts.WithDelegatedControl,
+		ntopts.WithDeletionPropagationPolicy(metadata.DeletionPropagationPolicyOrphan),
+		ntopts.RepoSyncPermissions(policy.CoreAdmin()))
+
+	nt.T.Log("Add all resources to a single RootSync")
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(rootSync)
+	nt.Must(rootSyncGitRepo.Add("acme/ns1.yaml", ns1()))
+	nt.Must(rootSyncGitRepo.Add("acme/anvil_crd.yaml", anvilCRD()))
+	nt.Must(rootSyncGitRepo.Add("acme/anvil_cr.yaml", anvilCR()))
+	nt.Must(rootSyncGitRepo.Add("acme/cm1.yaml", cm1()))
+	nt.Must(rootSyncGitRepo.Add("acme/sa1.yaml", sa1()))
+	nt.Must(rootSyncGitRepo.CommitAndPush("Add objects"))
+	nt.Must(nt.WatchForAllSyncs())
+	// Validate metadata for each object
+	nt.Must(verifyManagedBy(nt, ns1(), rootSync))
+	nt.Must(verifyManagedBy(nt, anvilCRD(), rootSync))
+	nt.Must(verifyManagedBy(nt, anvilCR(), rootSync))
+	nt.Must(verifyManagedBy(nt, cm1(), rootSync))
+	nt.Must(verifyManagedBy(nt, sa1(), rootSync))
+
+	nt.T.Cleanup(func() {
+		rs := nomostest.RootSyncObjectV1Beta1FromRootRepo(nt, rootSync.Name)
+		nt.Must(nt.KubeClient.Apply(rs))
+		nt.Must(nt.WatchForAllSyncs())
+	})
+	nt.T.Log("Delete the single RootSync to orphan all resources")
+	nt.Must(nt.KubeClient.Delete(k8sobjects.RootSyncV1Beta1(rootSync.Name)))
+	nt.Must(nt.Watcher.WatchForNotFound(kinds.RootSyncV1Beta1(), rootSync.Name, rootSync.Namespace))
+	nt.Must(verifyUnmanaged(nt, ns1()))
+	nt.Must(verifyUnmanaged(nt, anvilCRD()))
+	nt.Must(verifyUnmanaged(nt, anvilCR()))
+	nt.Must(verifyUnmanaged(nt, cm1()))
+	nt.Must(verifyUnmanaged(nt, sa1()))
+
+	nt.T.Cleanup(func() {
+		// Remove objects from the original RSync for cleanup
+		repo := nt.SyncSourceGitReadWriteRepository(rootSync)
+		nt.Must(repo.Remove("acme/ns1.yaml"))
+		nt.Must(repo.Remove("acme/anvil_crd.yaml"))
+		nt.Must(repo.Remove("acme/anvil_cr.yaml"))
+		nt.Must(repo.Remove("acme/cm1.yaml"))
+		nt.Must(repo.Remove("acme/sa1.yaml"))
+		nt.Must(repo.CommitAndPush("Remove orphaned objects"))
+	})
+	nt.T.Log("Adopt each resource by various RootSyncs/RepoSyncs")
+	repo := nt.SyncSourceGitReadWriteRepository(rootSync1)
+	nt.Must(repo.Add("acme/ns1.yaml", ns1()))
+	nt.Must(repo.CommitAndPush("Take ownership of ns1 in root-sync-1"))
+	repo = nt.SyncSourceGitReadWriteRepository(rootSync2)
+	nt.Must(repo.Add("acme/anvil_crd.yaml", anvilCRD()))
+	nt.Must(repo.Add("acme/anvil_cr.yaml", anvilCR()))
+	nt.Must(repo.CommitAndPush("Take ownership of anvil in root-sync-2"))
+	repo = nt.SyncSourceGitReadWriteRepository(repoSync1)
+	nt.Must(repo.Add("acme/cm1.yaml", cm1()))
+	nt.Must(repo.CommitAndPush("Take ownership of cm1 in repo-sync-1"))
+	repo = nt.SyncSourceGitReadWriteRepository(repoSync2)
+	nt.Must(repo.Add("acme/sa1.yaml", sa1()))
+	nt.Must(repo.CommitAndPush("Take ownership of sa1 in repo-sync-2"))
+	// Ready check recreates the root-sync, which we don't want
+	nt.Must(nt.WatchForAllSyncs(nomostest.SkipReadyCheck(), nomostest.SkipRootRepos(rootSync.Name)))
```

Comment:
Will this do anything? Doesn't SkipRootRepos skip the RootSync check?

Reply:
It skips based on the list of names that are passed in. It still waits for the other RootSyncs and RepoSyncs to be synced